pkg/azureclients/vmssvmclient/azure_vmssvmclient.go

```diff
@@ -329,21 +329,23 @@ func (c *Client) updateVMSSVM(ctx context.Context, resourceGroupName string, VMS
 	)
 
 	response, rerr := c.armClient.PutResource(ctx, resourceID, parameters)
+	result := compute.VirtualMachineScaleSetVM{}
 	defer c.armClient.CloseResponse(ctx, response)
 	if rerr != nil {
 		klog.V(5).Infof("Received error in %s: resourceID: %s, error: %s", "vmssvm.put.request", resourceID, rerr.Error())
-		return rerr
+		return result, rerr
 	}
 
 	if response != nil && response.StatusCode != http.StatusNoContent {
 		_, rerr = c.updateResponder(response)
 		if rerr != nil {
 			klog.V(5).Infof("Received error in %s: resourceID: %s, error: %s", "vmssvm.put.respond", resourceID, rerr.Error())
-			return rerr
+			return result, rerr
```

Comment:
need to parse response, this is still an empty result here: `compute.VirtualMachineScaleSetVM{}`

Reply:
fixed